Global/JetBrains.gitignore

```diff
@@ -44,6 +44,16 @@ cmake-build-*/
 # IntelliJ
 out/
 
+# Eclipse
+.classpath
+.project
+.settings/
+
+# Intellij
+.idea/
```

Comment:
The start of this file has a set of more explicit exclusions inside `.idea/` for things that should not be version controlled, and this rule here will ensure everything else is excluded. I'm still in favour of committing things that can be shared, based on the guidance from JetBrains here: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839